src/databases/functions/users/user-set-karma-score.js

```diff
@@ -0,0 +1,15 @@
+const {User} = require('../../models');
+
+module.exports = async (userId, combined_user_score, karma_score) => {
+  if (!userId) return null;
+  try {
+    const result = await User.update(
+      {combined_user_score, karma_score},
+      {where: {user_id: userId}}
+    );
+    return result;
+  } catch (error) {
+    console.error(`Failed to set karma score user with id ${userId}: `, error);
+    return null;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling here only logs the error and returns `null`. This could make it difficult to handle errors upstream because the calling function won't know whether the operation failed or there was no `userId` provided. Consider throwing the error after logging it, or returning a more descriptive error message.

```diff
- console.error(`Failed to set karma score user with id ${userId}: `, error);
- return null;
+ console.error(`Failed to set karma score for user with id ${userId}: `, error);
+ throw error;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->